packages/server/modules/gatekeeper/repositories/workspaceSeat.ts

```diff
@@ -136,3 +141,67 @@ export const getWorkspaceRoleAndSeatFactory =
     })
     return rolesAndSeats[userId]
   }
+
+export const getWorkspacesUsersSeatsFactory =
+  (deps: { db: Knex }): GetWorkspacesUsersSeats =>
+  async (params) => {
+    const { requests } = params
+    const q = tables.workspaceSeats(deps.db).whereIn(
+      [WorkspaceSeats.col.workspaceId, WorkspaceSeats.col.userId],
+      requests.map(({ userId, workspaceId }) => [workspaceId, userId])
+    )
+    const results = await q
+
+    return results.reduce((acc, seat) => {
+      const { userId, workspaceId } = seat
+      if (!acc[workspaceId]) {
+        acc[workspaceId] = {}
+      }
+
+      if (!acc[workspaceId][userId]) {
+        acc[workspaceId][userId] = seat
+      }
+      return acc
+    }, {} as Awaited<ReturnType<GetWorkspacesUsersSeats>>)
+  }
+
+export const getProjectsUsersSeatsFactory =
+  (deps: { db: Knex }): GetProjectsUsersSeats =>
+  async (params: {
+    requests: Array<{
+      userId: string
+      projectId: string
+    }>
+  }) => {
+    const { requests } = params
+    const idPairs = requests.map(({ userId, projectId }) => [userId, projectId])
+
+    const q = tables
+      .streamAcl(deps.db)
+      .whereIn([StreamAcl.col.userId, StreamAcl.col.resourceId], idPairs)
```

Comment:
`WHERE (userId, resourceId) in ((x, y), (x, y), (x,y))` SQL syntax so we can get results for multiple workspaceIds with 1 SQL query